server/src/main/java/org/elasticsearch/search/aggregations/pipeline/movfn/MovingFunctions.java

```diff
@@ -0,0 +1,360 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.pipeline.movfn;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.ToDoubleFunction;
+
+/**
+ * Provides a collection of static utility methods that can be referenced from MovingFunction script contexts
+ */
+public class MovingFunctions {
+
+    /**
+     * Find the maximum value in a window of values.
+     * If all values are missing/null/NaN, the return value will be NaN
+     */
+    public static double windowMax(Collection<Double> values) {
```

Comment:
What do you think of using DoubleStream as the type here? I think it could save boxing in some cases?